### PR TITLE
Migrate from Flask to FastAPI

### PR DESCRIPTION
**Pull request by Lasall**
* See #119

I tried to keep the changes (almost) minimal. The tests are passing but manual testing is required. To avoid unnecessary errors, I changed the input names to the constructor arguments and moved most of them into a pydantic model class.
In some cases (PVAkku, EAutoAkku), different defaults were used, so I created inherited models, not sure if thats the pydantic way.

FastAPI provides input validation (pydantic) and generates a swagger interface automatically when starting the server. ~~The API should probably be exported to github pages or so (e.g. openapi specs).~~

### Todo:
- [ ] someone with knowledge of the system has to validate its functionality
- [x] update single_test_optimization.py
- [x] pydantic in requirements.txt and decision if version 1 or 2 (currently 2)
- [x] API is linked at swagger.io with spec from this repo and added to spynx

### CODE REVIEW

**Comment by NormannK** [comment]
this looks like a lot of work. I'm currently producing tests for the classes we don't have any yet. 
Do you think we should wait for this to push until we have all tests?

On the other hands the already successful tests do look very good.

**Comment by Lasall** [comment]
More tests make it probably safer.

**Comment by Lasall** [comment]
Could fix another bug with the help of single_test_optimization.py. We should probably add more calculation tests. Some sample requests with old and new server result in the same output (with the pdf it's fortunately easy to check) so from my side this can be thoroughly tested now.

**Comment by NormannK** [comment]
I thought about hashing the pdf and checking it vs the new version. what do you think?

**Comment by Lasall** [comment]
Since the pdf is generated from the optimize output, it would be enough to check that step imho.

**Comment by drbacke** [comment]
Thanks, i will check that in my local system 👍 
I will need some days, because i Work on ac Charge, which is wuite a Lot to change

**Comment by Lasall** [comment]
Moved the README info directly into the code (will be exported with Swagger API-doc). I noticed the `simulation_data` that is just a copy of `result`. Do we want to fill that in the future or can I remove it? (atm I disabled that code)

File: tests/test_openapi.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+DIR_PROJECT_ROOT = Path(__file__).parent.parent
+DIR_TESTDATA = Path(__file__).parent / "testdata"
+
+
+def test_openapi_spec_current():
+    """Verify the openapi spec hasn´t changed."""
+
+    old_spec_path = DIR_PROJECT_ROOT / "openapi.json"
+    new_spec_path = DIR_TESTDATA / "openapi-new.json"
+    generate_openapi(new_spec_path)
```
Comment (b0661) [inline tests/test_openapi.py:15]: Could you use formatted json for openapi(-new).json?

I changed a comment in the source that ended up in the openapi-new.json. It is very hard to find the culprit in a one line diff that basically tells you the whole line changed.

Comment (Lasall) [reply tests/test_openapi.py:15]: Sorry about that, spec is now pretty printed.

**Comment by drbacke** [comment]
Please re-run all tests, code changed quite a lot

**Comment by NormannK** [comment]
personally id like to push the PR #141 first. im currently looking into a failed test in main. after that i might be able to merge 141

**Comment by NormannK** [comment]
@Lasall done

**Comment by Lasall** [comment]
Some endpoints (/strompreis) are broken (same behavior on main) or are missing complete pydantic API information (e.g. /gesamtlast). Tests and /optimize, /pdf endpoints seem to be ok.

**Comment by NormannK** [comment]
its bad we don't have tests for them. what do you think? Migrate to Flask or fix the bugs first?

File: openapi.json [hunk 2]
```diff
@@ -0,0 +1,940 @@
+{
```
Comment (b0661) [inline openapi.json:1]: You may use https://github.com/Akkudoktor-EOS/EOS/blob/main/docs/akkudoktoreosserver/openapi.json for testing. This way the documentation will stay in sync with the server api.

**Comment by Lasall** [comment]
I moved the server entrypoint to akkudoktoreos. However I still fully removed flask as we can probably use the archived file in git for reference. The logic within the endpoints was mostly untouched besides data validation.

File: README.md [hunk 3]
```diff
@@ -111,250 +119,7 @@ These classes work together to enable a detailed simulation and optimization of
...
-- **Type**: null or object
-
-### start_solution
-
-- **Description**: An array of binary values (0 or 1) representing a possible starting solution for the simulation.
-- **Type**: Array
-- **Element Type**: Integer (0 or 1)
-- **Length**: 48
+See the Swagger documentation for detailed information: [EOS OpenAPI Spec](https://petstore3.swagger.io/?url=https://raw.githubusercontent.com/Akkudoktor-EOS/EOS/refs/heads/main/openapi.json)
+>>>>>>> 4b4cbf2 (Move API doc from README to pydantic model classes (swagger))
```
Comment (b0661) [inline README.md:125]: Seems to be a leftover.

**Comment by NormannK** [comment]
@Lasall @b0661 as soon as you both its ready to merge ill click the button.

**Comment by b0661** [comment]
@NormannK That’s fine with me.

**Comment by Lasall** [comment]
Ready to merge for me too. I looked over the MR and compared the flask side-by-side as the diff is not so good (e.g. different indentation). Found one smaller merge mistake (optimization test data generation), see last commit.

**Comment by NormannK** [comment]
Ill merge it this evening. I made PR'S for the dead endpoints. Ill adjust it after

**Review by NormannK — APPROVED** [review]

**Pull request merged**